spec.html

```diff
@@ -32903,6 +32903,22 @@ <h1>Properties of the String Prototype Object</h1>
         1. Throw a *TypeError* exception.
       </emu-alg>
 
+      <emu-clause id="sec-string.prototype.at">
+        <h1>String.prototype.at ( _index_ )</h1>
+        <emu-alg>
+          1. Let _O_ be ? RequireObjectCoercible(*this* value).
+          1. Let _S_ be ? ToString(_O_).
+          1. Let _len_ be the length of _S_.
+          1. Let _relativeIndex_ be ? ToIntegerOrInfinity(_index_).
+          1. If _relativeIndex_ &ge; 0, then
+            1. Let _k_ be _relativeIndex_.
+          1. Else,
+            1. Let _k_ be _len_ + _relativeIndex_.
+          1. If _k_ &lt; 0 or _k_ &ge; _len_, return *undefined*.
+          1. Return the String value of length 1, containing the code unit at index _k_ in _S_.
```

Comment:
i wonder if both this and charAt could/should use `StringGetOwnProperty`?

Reply:
I don't think that adds any more clarity than this sentence, since none of the checking steps in StringGetOwnProperty are needed. But that points out that we should make the language to make a 1-length string consistent across all these uses, I thought it was just the two methods.

Reply:
Perhaps a simpler AO that this, charAt, and StringGetOwnProperty could call? I'm lightly hoping for something that doesn't rest on prose.

Reply:
Nothing wrong with using a consistent form in prose instead of an AO.

Reply:
It certainly works, but it's less clear to me that way.

Reply:
In this case it bottoms out at prose no matter what. StringGetOwnProperty bottoms out on prose. Unless you want to fully model strings as having some internal List of code units, which is not a project I think would add much right now.

Reply:
Fair point. It would certainly minimize the number of places that prose exists tho :-p yalls call either way, ofc.

Reply:
You could use `the substring of _S_ from _k_ to _k_ + 1`. (And similarly in StringGetOwnProperty and charAt.)

Reply:
I like that suggestion, @jmdyck, given its prevalence in other string algorithms.